java-checks/src/test/java/org/sonar/java/checks/serialization/RecordSerializationIgnoredMembersCheckTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.serialization;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.java.checks.verifier.CheckVerifier;
+
+import static org.sonar.java.checks.verifier.TestUtils.testSourcesPath;
+
+class RecordSerializationIgnoredMembersCheckTest {
+
+  @Test
+  void test() {
+    CheckVerifier.newVerifier()
+      .onFile(testSourcesPath("checks/serialization/RecordSerializationIgnoredMembersCheck.java"))
+      .withCheck(new RecordSerializationIgnoredMembersCheck())
+      .withJavaVersion(16)
```

Comment:
It may not be necessary to set the version explicitly as parsing the code with versions prior to 16 will not generate any issue

Reply:
Well, records are only available with 16 and we anyway parse with this version. So indeed, I'll drop the line. However, we still parse and raise issues. If I force a version below 16, it crash with a parsing issue.